### PR TITLE
feat(ci): GitHub Action for updating IC replica

### PR DESCRIPTION
**Pull request by smallstepman**
# Description

Workflow for updating IC replica. The workflow produces a PR that has to be reviewed. The PR contains the text which needs to be inserted into the changelog manually (also provides the link to edit the changelog file directly on the branch of created PR). 

Example of the workflow run output: https://github.com/smallstepman/sdk/runs/7522842430
Example of the PR created by the workflow: https://github.com/smallstepman/sdk/pull/39

---

The list of links to proposals is generated based on replica version sha used in the latest GitHub Release (not draft or prerelease). Currently, the workflow searches for `replica version used: ([a-f0-9]+)`, for example:
image
We would need to add this.

---
 
The workflow has to be run manually. GitHub Actions have a nice UI to input all necessary variables.
image
- when `latest` is selected, the replica version is fetched from https://ic-api.internetcomputer.org/api/v3/subnet-replica-versions?limit=50&offset=0
- when `custom` is selected, the SHA has to be provided manually
- option to include NNS canisters rebuild (used for e2e testing)
---
The workflow at its core still uses Nix and niv and `scripts/update-replica.sh`, though small tweaks to the scripts were necessary. 

**New dependency**: GitHub Action step which installs Nix (pinned to the commit of the latest release).

---

The workflow configures the author/committer/user in git client in a way, that creates pretty looking git commits 
image

---

_background: I struggled to install nix on my system due to errors, therefore decided to build a workflow for the part of CI which requires Nix. I'd love to use this workflow as a starting point for automating other chores, like updating motoko or ic-ref versions._

# How Has This Been Tested?

Ran many times on fork repo.

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I have edited the CHANGELOG accordingly.
- [ ] I have made corresponding changes to the documentation.

### CODE REVIEW

**Comment by ericswanson-dfinity** [comment]
Cool!  Can you update https://github.com/dfinity/sdk/blob/master/.github/CONTRIBUTING.adoc#updating-the-replica with instructions on how to kick off the workflow, an appropriate github URL, that sort of thing?

**Comment by smallstepman** [comment]
sure, done! heads up: the link in CONTRIBUTING.adoc will remain broken until this gets merged

File: .github/CONTRIBUTING.adoc [hunk 1]
```diff
@@ -109,6 +109,8 @@ To update Motoko to a given $VERSION from the motoko and motoko-base repos, exec
 ----
 # Requires niv to run. To install niv, run nix-env -iA nixpkgs.niv
 ./scripts/update-motoko.sh $VERSION
+# if you want to also ledger canisters for e2e tests, use:
```
Comment (ericswanson-dfinity) [inline .github/CONTRIBUTING.adoc:112]:
```suggestion
# if you want to also update ledger canisters for e2e tests, use:
```

File: .github/CONTRIBUTING.adoc [hunk 2]
```diff
@@ -87,6 +87,14 @@ What that means is your PR title should start with one of the following prefix:
 
 === Updating the Replica
 
+==== Using GitHub Action
+. Head over to https://github.com/dfinity/sdk/actions/workflows/update-replica-version.yml[the GitHub Aciton].
```
Comment (ericswanson-dfinity) [inline .github/CONTRIBUTING.adoc:91]:
```suggestion
. Head over to https://github.com/dfinity/sdk/actions/workflows/update-replica-version.yml[the GitHub Action].
```

**Review by ericswanson-dfinity — APPROVED** [review]
Just a couple typos left.  Looks great!

**Comment by smallstepman** [comment]
thanks!

**Pull request merged**